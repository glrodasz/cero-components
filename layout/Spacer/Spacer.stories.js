import { Spacer, options, styles } from '.'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Spacer, styles)
const ListTemplate = getListTemplate(Spacer, styles)

export default {
  title: 'Layout/Spacer/Default',
  component: Spacer,
  args: {
    isVisible: true,
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
  },
  parameters: { __sb: { fd: 'row' } },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
