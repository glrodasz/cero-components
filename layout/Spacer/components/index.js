export { default as Vertical } from './Vertical'
export { default as Horizontal } from './Horizontal'
