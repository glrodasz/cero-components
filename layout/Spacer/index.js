export { default, Spacer } from './Spacer'
export { options } from './constants'
export { default as styles } from './Spacer.module.css'
