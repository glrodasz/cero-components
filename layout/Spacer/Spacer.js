import React from 'react'
import PropTypes from 'prop-types'

import { Horizontal, Vertical } from './components'

import styles from './Spacer.module.css'
import { mapSize } from './helpers'
import withStyles from '../../hocs/withStyles'

export const Spacer = ({ getStyles, size, isVisible }) => (
  <div
    className={getStyles('spacer', {
      'is-visible': isVisible,
    })}
    style={{
      display: 'inline-block',
      width: mapSize(size),
      height: mapSize(size),
    }}
  />
)

Spacer.propTypes = {
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  getStyles: PropTypes.func.isRequired,
  isVisible: PropTypes.bool,
}

Spacer.defaultProps = {
  size: 'none',
  isVisible: false,
  getStyles: () => {},
}

Spacer.Horizontal = Horizontal
Spacer.Vertical = Vertical

export default withStyles(styles)(Spacer)
