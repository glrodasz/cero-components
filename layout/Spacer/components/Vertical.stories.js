import Spacer, { options } from '..'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Spacer.Vertical)
const ListTemplate = getListTemplate(Spacer.Vertical)

export default {
  title: 'Layout/Spacer/Vertical',
  component: Spacer.Vertical,
  args: {
    isVisible: true,
    height: 100,
    maxHeight: 100,
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
    height: { control: 'number' },
    maxHeight: { control: 'number' },
  },
  parameters: { __sb: { fd: 'row' } },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
