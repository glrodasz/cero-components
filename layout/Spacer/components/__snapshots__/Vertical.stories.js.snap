// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Spacer/Vertical Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 0,
      }
    }
  />
</div>
`;

exports[`Storyshots Layout/Spacer/Vertical Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 0,
      }
    }
  />
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 5,
      }
    }
  />
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 15,
      }
    }
  />
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 20,
      }
    }
  />
  <div
    className="spacer vertical is-visible"
    style={
      Object {
        "height": 100,
        "maxHeight": 100,
        "width": 40,
      }
    }
  />
</div>
`;
